piIOComm.c

```diff
@@ -311,7 +311,9 @@ INT32S piIoComm_sendRS485Tel(INT16U i16uCmd_p, INT8U i8uAddress_p,
 	ret = pibridge_req_gate_tmt(i8uAddress_p, i16uCmd_p, pi8uSendData_p,
 				    i8uSendDataLen_p, pi8uRecvData_p,
 				    rcvlen, timeout);
-	if (ret) {
+	if (ret != rcvlen) {
+		if (ret >= 0)
+			ret = -EIO;
```

Comment:
I've looked at all the callers of `piIoComm_sendRS485Tel()` and they either only check that the return value is non-zero or they ignore the return value.

In particular, all the callers in <a href="https://github.com/RevolutionPi/piControl/blob/ab3b62822abed8933ffeaa1a7e6eef55768cb062/RevPiDevice.c">RevPiDevice.c</a> only check for a non-zero return value. Curiously, they all store the return value as an *unsigned* int even though `piIoComm_sendRS485Tel()` returns a signed int. Very odd.

The single caller in <a href="https://github.com/RevolutionPi/piControl/blob/ab3b62822abed8933ffeaa1a7e6eef55768cb062/PiBridgeMaster.c">PiBridgeMaster.c</a> stores the return value in `piCore_g.statusGateTel`. At least that's a signed integer. But the `KB_CONFIG_SEND` ioctl, which is the only one sending userspace gate telegrams, stores the return value in <a href="https://github.com/RevolutionPi/piControl/blob/ab3b62822abed8933ffeaa1a7e6eef55768cb062/piControlMain.c#L1501">status</a>, then unconditionally overwrites it with 0. So userspace never learns if sending the gate telegram failed. This looks like another bug, oh dear! :(